dwave/optimization/states.pyx

```diff
@@ -216,16 +247,112 @@ cdef class States:
             version:
                 A 2-tuple indicating which serialization version to use.
 
-        TODO: describe the format
+        Format Specification (Version 1.0):
+
+            The first section of the file is the header, as described in
+            :meth:`Model.into_file()`.
+
+            Following the header, the remaining data is encoded as a zip file.
+            All arrays are saved using the NumPy serialization format, see
+            :func:`numpy.save()`.
+
+            The information in the header is also saved in a json-formatted
+            file ``info.json``.
+
+            The serialization version is saved in a file ``version.txt``.
+
+            The states have the following structure.
```

Comment:
This should note that only the states of decisions and other non-deterministic symbols is saved. I will fix.

Reply:
the code below for v1.0 uses `only_decision=False`

Reply:
Yes, because it's using the `if symbol.deterministic_state(): continue` logic instead. Before we just hard-limited it to decisions which gave the same thing if you don't have non-determistic intermediate nodes (which we currently do not, but want to add).